pkg/nsx/services/vpc/builder.go

```diff
@@ -53,28 +53,31 @@ func combineVPCIDAndLBSID(vpcID, lbsID string) string {
 }
 
 func buildNSXVPC(obj *v1alpha1.NetworkInfo, nsObj *v1.Namespace, nc v1alpha1.VPCNetworkConfiguration, cluster string,
-	nsxVPC *model.Vpc, useAVILB bool, lbProviderChanged bool) (*model.Vpc, error) {
+	nsxVPC *model.Vpc, useAVILB bool, lbProviderChanged bool, serviceClusterReady bool) (*model.Vpc, error) {
 	vpc := &model.Vpc{}
+	// enable LB Endpoint for either AVI LB or day2 DTGW
+	enableLBEndpoint := useAVILB || serviceClusterReady
 	if nsxVPC != nil {
+		oldEnableLBEndpoint := (nsxVPC.LoadBalancerVpcEndpoint != nil) && (nsxVPC.LoadBalancerVpcEndpoint.Enabled != nil) && *nsxVPC.LoadBalancerVpcEndpoint.Enabled
 		// for upgrade case, only check public/private ip block size changing
```

Comment:
this comment needs to be change according the following logics

Reply:
Updated, thanks